deepspeed/runtime/zero/stage2.py

```diff
@@ -1376,22 +1383,25 @@ def buffered_reduce_fallback(self,
 
     # views the tensor as multiple partitions and returns
     # those partitions
-    def get_data_parallel_partitions(self, tensor, group_id):
+    def get_data_parallel_partitions(self, tensor, group_id, alignment_factor):
```

Comment:
I think this function does not need to be modified.

Reply:
This function is what does the 'alignment'. 

To be specific, the nccl alignment we are concerned about here is the alignment of the start location of each partition. See the newly added testcases.

Reply:
Yes I get that but isn't it redundant since the alignment is already done by flatten_dense_tensor_aligned?